format/FormatNexusEigerDLS.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import absolute_import, division, print_function
+
+import ast
+
+import h5py
+
+from dxtbx.format.FormatNexus import FormatNexus
+
+
+def get_count_limit_from_meta(meta_file_name):
+    with h5py.File(meta_file_name, "r") as f:
+
+        config = f["/config"][()]
+        config_data = ast.literal_eval(config.decode("utf-8"))
+
+    return config_data["countrate_correction_count_cutoff"]
+
+
+def get_bit_depth_from_meta(meta_file_name):
+    with h5py.File(meta_file_name, "r") as f:
+
+        config = f["/config"][()]
+        config_data = ast.literal_eval(config.decode("utf-8"))
+
+    return config_data["bit_depth_image"]
+
+
+class FormatNexusEigerDLS(FormatNexus):
+    @staticmethod
+    def understand(image_file):
+        # Get the file handle
+        with h5py.File(image_file, "r") as handle:
+            name = FormatNexusEigerDLS.get_instrument_name(handle)
+            if name is None or name.lower() not in (b"i03", b"i04", b"vmxi"):
+                return False
+
+        return True
+
+    def __init__(self, image_file, **kwargs):
+        """Initialise the image structure from the given file."""
+
+        super(FormatNexusEigerDLS, self).__init__(image_file, **kwargs)
+        try:
+            if self._image_file.endswith("_master.h5"):
+                meta = self._image_file.replace("_master.h5", "_meta.h5")
+            elif self._image_file.endswith(".nxs"):
+                meta = self._image_file.replace(".nxs", "_meta.h5")
```

Comment:
Can `meta` be saved as `self._meta` (outside of the try/except) so that it can be re-used below in `get_detector` rather than duplicating the code?

Reply:
+ 1

In fact, the search for the meta file here is weak anyway as ideally it should be searching for the `ExternalLink` in the file to identify the meta file name, not using "rules" - however 30 minutes down the 🐇🕳  has not got me an answer yet on this one...

Reply:
Resolved in 185529189bbbee2b8852a5c02dc4c2b163c92dca